### PR TITLE
SW-2708 initialize org time zone on login

### PR DESCRIPTION
**Pull request by karthikbtf**
- initialize org time zone on login, show message that it was initialized on every admin login (until admin clicks the 'x' or overrides the org tz) - state is captured in a user preference
- move user time zone initialization into same place org time zone is initialized (show user tz initialized message every time until the 'x' is clicked or user overrides the tz)
- show combined messaging for user/org (see screenshots and [JIRA](https://terraformation.atlassian.net/browse/SW-2708))
- cleaned up const/variable names in organization provider
- fix callback on page-message close, which wasn't being called
- this is my third attempt in finding the right place for the combined tz initialization logic 
- we will have to do something similar for weight units i18n... will do more refactor at that point with a dedicated service (move code out of App)

Terraware App 2023-01-18 14-29-49

Terraware App 2023-01-18 10-15-02

Terraware App 2023-01-18 10-02-09

### CODE REVIEW

**Review by chudo76 — APPROVED** [review]
LGTM!

**Review by charlie-tf — COMMENTED** [review]
This looks good. Nice work! Just one minor question.

File: src/App.tsx [hunk 1]
```diff
@@ -239,10 +254,89 @@ function AppBase() {
...
+      }
+
+      let message: string | string[] = '';
+
+      if (notifyUser && notifyOrg) {
+        message = strings.formatString(
+          strings.TIME_ZONE_INITIALIZED_USER_ORG_MESSAGE,
+          getTimeZone(timeZones, userTz.timeZone)?.longName,
+          (<Link to={APP_PATHS.ORGANIZATION}>{strings.ORGANIZATION}</Link>) as any,
+          <Link to={APP_PATHS.MY_ACCOUNT}>{strings.MY_ACCOUNT}</Link>
```
[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/App.tsx:277]: There was no warning here, let me try again and see why it complained.

**Comment by constanzauanini** [comment]
Looks good!! Just a question, does it appear only on the home page? If I go to other page and the return to home page I don't see the message anymore, but if I do a refresh the message comes back. Is that the expected behavior?

**Comment by karthikbtf** [comment]
> Looks good!! Just a question, does it appear only on the home page? If I go to other page and the return to home page I don't see the message anymore, but if I do a refresh the message comes back. Is that the expected behavior?

For now yes, this is the expected behavior.
We only want this on initial visit to app and not keep it around the entire session. 
Our page message component handles one message at a time.. we want to be able to show other messages as well. We need to redo our page message component at some point. Current design/ux needs some change as well.

**Review by charlie-tf — APPROVED** [review]
👍 Seems to be working well in Vercel also

**Comment by karthikbtf** [comment]
> 👍 Seems to be working well in Vercel also

Thanks, @charlie-tf - I have one fix to do, this notification should only happen when the org id changes,, need to figure that out somehow. Right now if you edit the org name for example.. the popup is triggered again. Sigh.
